### PR TITLE
Temperature sensor support

### LINKED ISSUE #568: temperature sensor support (cpu on stabilizer and lm75 on pounder)
* The CPU has an internal temperature sensor connected to ADC3. Implement that and add it to telemetry in `dual-iir`/`lockin`.
* Pounder has an LM75 connected via I2C. Implement that and add it to the Pounder API (to be added to application telemetry)

### PR DESCRIPTION
**Pull request by ryan-summers**
This PR fixes #568 by adding temperature sensor support for Pounder and Stabilizer's CPU.

The LM75 on Pounder is exposed as a public member variable, so end-user applications can choose where to use it.

The Stabilizer CPU temperature sensor is exposed as a member of `StabilizerDevices` and can be used to measure the CPU junction temperature.

### Testing
I installed Pounder on top of Stabilizer and modified `dual-iir` to report both stabilizer CPU and pounder temperatures in telemetry. Temperatures were measured over time using MQTT explorer:
image

### CODE REVIEW

File: src/bin/dual-iir.rs [hunk 1]
```diff
@@ -194,6 +194,7 @@ mod app {
         dacs: (Dac0Output, Dac1Output),
         iir_state: [[iir::Vec5<f32>; IIR_CASCADE_LENGTH]; 2],
         generator: FrameGenerator,
+        temp_sensor: stabilizer::hardware::temp_sensor::CpuTempSensor,
```
Comment (jordens) [inline src/bin/dual-iir.rs:197]:
```suggestion
        cpu_temp: stabilizer::hardware::temp_sensor::CpuTempSensor,
```

as it's not the only temperature sensor on Stabilizer.

File: src/bin/dual-iir.rs [hunk 2]
```diff
@@ -252,6 +253,7 @@ mod app {
             dacs: stabilizer.dacs,
             iir_state: [[[0.; 5]; IIR_CASCADE_LENGTH]; 2],
             generator,
+            temp_sensor: stabilizer.temperature_sensor,
```
Comment (jordens) [inline src/bin/dual-iir.rs:256]:
```suggestion
            cpu_temp: stabilizer.cpu_temp,
```

File: src/bin/dual-iir.rs [hunk 3]
```diff
@@ -427,7 +429,7 @@ mod app {
         c.shared.network.lock(|net| net.direct_stream(target));
     }
 
-    #[task(priority = 1, shared=[network, settings, telemetry])]
+    #[task(priority = 1, shared=[network, settings, telemetry], local=[temp_sensor])]
```
Comment (jordens) [inline src/bin/dual-iir.rs:432]:
```suggestion
    #[task(priority = 1, shared=[network, settings, telemetry], local=[cpu_temp])]
```

File: src/bin/dual-iir.rs [hunk 4]
```diff
@@ -438,8 +440,11 @@ mod app {
             .lock(|settings| (settings.afe, settings.telemetry_period));
 
         c.shared.network.lock(|net| {
-            net.telemetry
-                .publish(&telemetry.finalize(gains[0], gains[1]))
+            net.telemetry.publish(&telemetry.finalize(
+                gains[0],
+                gains[1],
+                c.local.temp_sensor.get_temperature().unwrap(),
```
Comment (jordens) [inline src/bin/dual-iir.rs:446]:
```suggestion
                c.local.cpu_temp.get_temperature().unwrap(),
```

etc

File: src/hardware/mod.rs [hunk 5]
```diff
@@ -11,6 +11,7 @@ pub mod pounder;
 pub mod setup;
 pub mod shared_adc;
 pub mod signal_generator;
+pub mod temp_sensor;
```
Comment (jordens) [inline src/hardware/mod.rs:14]:
```suggestion
pub mod cpu_temp;
```

**Review by jordens — CHANGES_REQUESTED** [review]
Just a small rename.
I do wonder why the measured CPU temperature is fluctuating that much.
https://github.com/quartiq/stabilizer/issues/571 ist certainly a factor but usually that's a constant error.

**Comment by ryan-summers** [comment]
> Just a small rename. I do wonder why the measured CPU temperature is fluctuating that much. #571 ist certainly a factor but usually that's a constant error.

Is it usually constant? I imagine running at above the rated max clock rate would definitely result in fluctuation. Let me also take a second look at the required settling time specified in the datasheet.

**Comment by jordens** [comment]
Few things are constant, but I'd be surprised if our CPU temperature changes by more than 0.1 K in a second when running.
Let's see how Norman's ADC changes pan out. They will lower the ADC clock (but also lower the sampling time). We may have to set up a different sampling time (entire ADC configuration) for the temperature sensor.

**Pull request merged**